### PR TITLE
Gaspricer support apollo

### PR DESCRIPTION
**Pull request by giskook**
Closes #<issue number>.

### What does this PR do?
Support apollo

### Reviewers

Main reviewers:

- @John
- @Doe

Codeowner reviewers:

- @-Alice
- @-Bob

### CODE REVIEW

File: config/apollo/apollo.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	if c == nil {
+		return false
+	}
+	namespaces := strings.Split(c.config.Apollo.NamespaceName, ",")
+	for _, namespace := range namespaces {
+		cache := c.GetConfigCache(namespace)
+		cache.Range(func(key, value interface{}) bool {
+			loaded = true
+			switch namespace {
+			case L2GasPricer:
```
Comment (louisliu2048) [inline config/apollo/apollo.go:60]: can we use only one apollo client to support all the other params which need dynamic config?

Comment (giskook) [reply config/apollo/apollo.go:60]: sure, for now, a service may have two namespace, the first one `L2GasPricer` is a txt format, which include all config, the other one is only for halt the service.

Comment (giskook) [reply config/apollo/apollo.go:60]: > can we use only one apollo client to support all the other params which need dynamic config?
now every service have it's own configure. the configure dont share between services.

File: config/apollo/apollo.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+type CustomChangeListener struct {
+	*Client
+}
+
+// OnChange is the change listener
+func (c *CustomChangeListener) OnChange(changeEvent *storage.ChangeEvent) {
+	for key, value := range changeEvent.Changes {
+		if value.ChangeType == storage.MODIFIED {
+			switch changeEvent.Namespace {
+			case L2GasPricerHalt:
```
Comment (zjg555543) [inline config/apollo/apollo.go:79]: What's the halt mean? Does it halt the apollo client

File: config/apollo/namespace.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package apollo
+
+const (
+	// L2GasPricer is the l2gaspricer namespace,the content of the namespace is the config for l2gaspricer with toml format
+	L2GasPricer = "l2gaspricer.txt"
+	// L2GasPricerHalt is the l2gaspricer halt namespace,change the Halt to different value will halt l2gaspricer
+	L2GasPricerHalt = "l2gaspricer"
```
Comment (zjg555543) [inline config/apollo/namespace.go:7]: Maybe modify to "l2gaspricerHalt"?

File: config/apollo/apollo.go (hunk 2 above)
Comment (giskook) [reply config/apollo/apollo.go:79]: halt the service

File: config/apollo/namespace.go (hunk 3 above)
Comment (giskook) [reply config/apollo/namespace.go:7]: got it

**Review by zjg555543 — APPROVED** [review]

**Pull request merged**